core/cont/inc/TCollection.h

```diff
@@ -296,78 +298,20 @@ namespace Internal {
 const TCollection &EmptyCollection();
 bool ContaineeInheritsFrom(TClass *cl, TClass *base);
 
-/// @brief Internal help class implmenting an iterator for TRangeDynCast.
-template <class Containee> // Containee must derive from TObject.
-class TRangeDynCastIterator : public TIter {
-   static_assert(std::is_base_of<TObject, Containee>::value, "Containee type must inherit from TObject");
-
-   /// This is a workaround against ClassDefInline not supporting classes
-   /// missing their default constructor or having them private.
-   template <class T>
-   friend class ROOT::Internal::ClassDefGenerateInitInstanceLocalInjector;
-
-   TRangeDynCastIterator() = default;
-
-public:
-   using TIter::TIter;
-   TRangeDynCastIterator(const TIter &iter) : TIter(iter) {}
-
-   Containee *operator()() = delete;
-
-   Containee *Next() { return dynamic_cast<Containee *>(TIter::Next()); }
-   Containee *operator*() const { return dynamic_cast<Containee *>(TIter::operator*()); }
-
-   ClassDefInline(TRangeDynCastIterator, 0);
-};
-
 } // namespace Internal
 
 namespace Detail {
 
-/// @brief TTypedIter is a typed version of TIter.
-///
-/// This requires the collection to contains elements of the type requested
-/// (or a derived class).  Any deviation from this expectation
-/// will only be caught/reported by an assert in debug builds.
-///
-/// This is best used with a TClonesArray, for other cases prefered TRangeDynCast.
-///
-/// The typical use is:
-/// ```{.cpp}
-///    TTypedIter<TBaseClass> next(tbaseClassClonesArrayPtr);
-///    while(auto bcl = next()) {
-///       ... use bcl as a TBaseClass*
-///    }
-/// ```
-template <class Containee> // Containee must derive from TObject.
-class TTypedIter : public TIter {
```

Comment:
Is this breaking code that's somewhere out in the world? Or is this unlike, because it's internal?

Reply:
Unlikely, because it was not even used in ROOT, but yes I guess it's better to keep TTypedIter because it's not in the internal namespace.